### PR TITLE
Techspec adjustments - Verify JWKS (EXPOSUREAPP-10194)

### PR DESCRIPTION
**Pull request by BMItr**
This PR updates to the latest spec changes for EXPSOUREAPP-10194. 
Additonal steps added, regarding Verify JWKs

### CODE REVIEW

**Review by mtwalli — APPROVED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt [hunk 1]
```diff
@@ -4,8 +4,27 @@ import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingErrorCode
...
+fun DccTicketingServiceIdentityDocument.verifyJwks(emptyX5cErrorCode: DccTicketingErrorCode) {
+    Timber.d("verifyJwks(emptyX5cErrorCode=%s)", emptyX5cErrorCode)
+
+    val hasAnyEmptyX5c = verificationMethod
+        .mapNotNull { it.publicKeyJwk }
+        .any { it.x5c.isEmpty() }
+
+    when (hasAnyEmptyX5c) {
+        true -> Timber.d("%s passed verification", this)
+        false -> throw DccTicketingException(errorCode = emptyX5cErrorCode)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt:24]: I think we should switch ,right?
```kotlin
when (hasAnyEmptyX5c) {
        false -> Timber.d("%s passed verification", this)
        true -> throw DccTicketingException(errorCode = emptyX5cErrorCode)
```

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt [hunk 2]
```diff
@@ -4,8 +4,27 @@ import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingErrorCode
...
 import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingServiceIdentityDocument
+import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingVerificationMethod
 import timber.log.Timber
 
+/**
+ * @throws [DccTicketingException] if there is any element in [DccTicketingServiceIdentityDocument.verificationMethod]
+ * where [DccTicketingVerificationMethod.publicKeyJwk] is set and [DccJWK.x5c] does not contain at least one element,
+ * the operation shall abort with the specified error code, otherwise the document is valid
+ */
+fun DccTicketingServiceIdentityDocument.verifyJwks(emptyX5cErrorCode: DccTicketingErrorCode) {
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt:15]: isn't the `emptyX5cErrorCode` always the same? or what's the reason to have it as a parameter here?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt:15]: They look similar but no , one starts with `VD` and other with `VS` 😄 , I had the same question

Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt:15]: :smile_cat:

Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/DccTicketingRequestServiceHelpers.kt:15]: for ValidationService ist it VS and for ValidationDecorator it is VD

**Review by mtwalli — APPROVED** [review]

**Pull request merged**